src/gluonts/dataset/common.py

```diff
@@ -207,13 +213,144 @@ def files(self) -> List[Path]:
+            ".jsonl.gz",
+        ]
+        if not any(path.name.endswith(ending) for ending in valid_endings):
+            return False
+        return not path.name.startswith(".")
+
+
+class ArrowDataset(Dataset):
+    _SOURCE_COL_NAME = "_GLUONTS_SOURCE"
+    _ROW_COL_NAME = "_GLUONTS_ROW"
+
+    def __init__(
+        self,
+        table: pa.Table,
+        freq: str,
+        chunk_size: int = 100,
+    ):
+        """
+
+        An on disk Dataset based on pyarrow tables that is faster than json for large datasets or
+        datasets with long time series.
+
+        Use `ArrowDataset.load_files` to load the dataset from disk and
+        `ArrowDataset.write_table_from_records` to save recors in arrow format.
+        """
+        self.table = table
+        self.chunk_size = chunk_size
+        self.freq = freq
+        self._process = ProcessDataEntry(self.freq, one_dim_target=True)
+        self.reader = ArrowReader(self.table, chunk_size=self.chunk_size)
+
+    @classmethod
+    def load_files(
+        cls,
+        paths: Union[str, Path, List[str], List[Path]],
+        freq: str,
+        chunk_size: int = 100,
+    ):
+        paths = util.resolve_paths(paths) if paths is not None else None
+        files = util.find_files(paths, lambda p: str(p).endswith(".arrow"))
+        assert (
+            len(files) > 0
+        ), f"Could not find any arrow files in paths: {paths}"
+        return MemmapArrowDataset(files, freq, chunk_size)
+
+    def __iter__(self) -> Iterator[DataEntry]:
+        row_number = 0
+        bounds = get_bounds_for_mp_data_loading(len(self))
+        for rec in self.reader.iter_slice(bounds.lower, bounds.upper):
+            if self._SOURCE_COL_NAME in rec:
+                rec["source"] = SourceContext(
+                    source=rec[self._SOURCE_COL_NAME],
+                    row=rec[self._ROW_COL_NAME],
+                )
+                del rec[self._SOURCE_COL_NAME]
+                del rec[self._ROW_COL_NAME]
+            else:
+                rec["source"] = SourceContext(
+                    source="arrow.Table", row=row_number
+                )
+            rec = self._process(rec)
+            yield rec
+            row_number += 1
+
+    def __len__(self):
+        return len(self.table)
+
+    @classmethod
+    def write_table_from_records(
+        cls,
+        it: Iterable[DataEntry],
+        file_path: Union[str, Path],
+        chunk_size=100,
+    ) -> None:
+        """
+        Write time series records as an arrow dataset to the file_path.
+        """
+        file_path = Path(file_path)
+        msg = f"Converting to arrow dataset: {file_path}"
+        l: Optional[int] = None
+        try:
+            l = len(it)  # type: ignore
+        except:
+            l = None
+        with ArrowWriter(file_path, chunk_size=chunk_size) as writer:
+            for rec in tqdm(it, total=l, desc=msg):
+                rec = rec.copy()
+                float_np_fields = [
```

Comment:
This looks somewhat problematic to me, since it appears we are only supporting a few fixed fields.

I'm fine with leaving it as it is, since we have it elsewhere too.

Reply:
Yes, the `ArrowWriter` is very general and can write numpy arrays, lists etc. But this part is specific for our stuff. It just makes sure that the data we write is in the right format. If we write the target as a list for example, it slows things down again when reading, since you have to convert it to an array.